Makefile

```diff
@@ -11,6 +12,7 @@ init:
 .PHONY: clean
 clean:
 	rm -rf $(BUILD_DIR)
+	rm -rf $(BIN_DIR)
```

Comment:
If you remove the bin dir as part of each clean, then I argue, you should make it a sub-directory of _BUILD_DIR_. Usually I am of the point of view, everything which gets installed/created during the build goes somewhere under build. A reason for not doing this, would be to have something which takes a long time to do (for example downloading the Vagrant boxes in CDK 2) and you want to cache it separately. If you want to do this, then you need two clean targets. Otherwise there is no use of this distinction. You could have for example 'clean' and 'clean-all'. It depends how you want to look at this. Right now, however, the potential purpose of "caching" the binary is not achieved.